src/test/java/uk/gov/pay/ledger/transaction/service/TransactionMetadataServiceTest.java

```diff
@@ -76,9 +84,36 @@ public void shouldNotTryToInsertMetadata() {
                 .withDefaultEventDataForEventType(SalientEventType.CAPTURE_SUBMITTED.name())
                 .toEntity();
 
-        service.upsertMetadataFor(paymentCreatedEvent);
+        EventDigest eventDigest = EventDigest.fromEventList(List.of(paymentCreatedEvent));
+
+        service.upsertMetadataFor(eventDigest);
 
         verify(mockMetadataKeyDao, never()).insertIfNotExist(anyString());
         verify(mockTransactionMetadataDao, never()).upsert(anyLong(), anyString(), anyString());
     }
+
+    @Test
+    public void shouldUpsertValue() {
+        String externalId = "transaction-id";
+        TransactionEntity transaction = aTransactionFixture().withState(TransactionState.STARTED).toEntity();
+        service = new TransactionMetadataService(mockMetadataKeyDao, mockTransactionMetadataDao, mockTransactionDao);
+
+        Event event = anEventFixture().withResourceType(PAYMENT)
+                .withResourceExternalId(externalId)
+                .withEventType("ADMIN_MANUALLY_DID_AN_UPDATE")
+                .withEventData(new GsonBuilder().create()
+                        .toJson(Map.of("external_metadata", Map.of("key1", true))))
+                .toEntity();
+
+        Event previousEvent = anEventFixture().withResourceType(PAYMENT)
+                .withEventType("USER_APPROVED_FOR_CAPTURE")
+                .toEntity();
+        EventDigest paymentEventDigest = EventDigest.fromEventList(List.of(event, previousEvent));
+
+        when(mockTransactionDao.findTransactionByExternalId(externalId)).thenReturn(Optional.of(transaction));
+
+        service.upsertMetadataFor(paymentEventDigest);
+
+        verify(mockTransactionMetadataDao).upsert(any(), anyString(), anyString());
```

Comment:
does it make sense to assert on the actual strings to make extra sure? don't see why we need to use anyString() here.